teapots/classic-teapot/src/main/cpp/CMakeLists.txt

```diff
@@ -18,35 +18,52 @@ cmake_minimum_required(VERSION 3.4.1)
 
 # build native_app_glue as a static lib
 add_library(native_app_glue STATIC
-    ${ANDROID_NDK}/sources/android/native_app_glue/android_native_app_glue.c)
-
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wall -fno-exceptions -fno-rtti")
+  ${ANDROID_NDK}/sources/android/native_app_glue/android_native_app_glue.c)
 
 # build the ndk-helper library
-set(ndk_helper_dir ../../../../common/ndk_helper)
-add_subdirectory(${ndk_helper_dir} ndk_helper)
-
-# Export ANativeActivity_onCreate(), 
-# Refer to: https://github.com/android-ndk/ndk/issues/381.
-set(CMAKE_SHARED_LINKER_FLAGS
-    "${CMAKE_SHARED_LINKER_FLAGS} -u ANativeActivity_onCreate")
+set(ndkHelperSrc ../../../../common/ndk_helper)
+add_subdirectory(${ndkHelperSrc} ndkHelperBin)
 
 # now build app's shared lib
-add_library(TeapotNativeActivity SHARED
+add_library(TeapotNativeActivity
+  SHARED
     TeapotNativeActivity.cpp
-    TeapotRenderer.cpp)
+    TeapotRenderer.cpp
+)
+set_target_properties(TeapotNativeActivity
+  PROPERTIES
+    CXX_STANDARD 11
+    CXX_STANDARD_REQUIRED YES
+    CXX_EXTENSIONS NO
+)
+target_compile_features(TeapotNativeActivity
+  PRIVATE cxx_noexcept
```

Comment:
You're not using the visibility of the cxx_noexcept feature consistently. In some targets you have it as PRIVATE, and in some as PUBLIC. Which one is better is a different debate, but at least you could be consistent about it.

Reply:
`cxx_noexcept` is not the same as `-fno-exceptions`. This is actually saying the opposite: this project requires that the `noexcept` keyword be available, which requires `-fexceptions`.

Reply:
yeah, for the cxx_noexcept, it is client of the compile flag -fno-exception. Will move it into the compile flag and wait for some better ways, thanks!